README.md

```diff
@@ -1,26 +1,123 @@
+* Portable
+  - Tested on Linux/MacOS
+  - Tested on x64, ARMv5
+* Small Install Size
+  - Sub 200k library size
+  - OpenSSL, libsrtp, libjsmn, libusrsctp and libwebsockets dependencies.
+
+## Build
+### Download
 To download run the following command:
 
 `git clone --recursive https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c.git`
 
-### 2. Configure
+You will also need to install `pkg-config` and `CMake` and a build enviroment
+
+### Configure
 Create a build directory in the newly checked out repository, and execute CMake from it.
 
 `mkdir -p amazon-kinesis-video-streams-webrtc-sdk-c/build; cd amazon-kinesis-video-streams-webrtc-sdk-c/build; cmake .. `
 
+We have provided an example of using GStreamer to capture/encode video, and then send via this library. This is only build if `pkg-config` finds
+GStreamer is installed on your system.
 
-*TODO* Document all the available options
+By default we download all the libraries from GitHub and build them locally, so should require nothing to be installed ahead of time.
+If you do wish to link to existing libraries you can use the following flags to customize your build.
 
-### 3. Build
+You can pass the following options to `cmake ..`.
+
+* `-DADD_MUCLIBC` -- Add -muclibc c flag
+* `-DBUILD_DEPENDENCIES` -- Whether or not to build depending libraries from source
+* `-DBUILD_OPENSSL` -- If building dependencies, whether or not building openssl from source
+* `-DBUILD_TEST=TRUE` -- Build unit/integration tests, may be useful for confirm support for your device. `./tst/webrtc_client_test`
+* `-DCODE_COVERAGE` --  Enable coverage reporting
+* `-DCOMPILER_WARNINGS` -- Enable all compiler warnings
+* `-DADDRESS_SANITIZER` -- Build with AddressSanitizer
+* `-DMEMORY_SANITIZER` --  Build with MemorySanitizer
+* `-DTHREAD_SANITIZER` -- Build with ThreadSanitizer
+* `-DUNDEFINED_BEHAVIOR_SANITIZER` Build with UndefinedBehaviorSanitizer`
+
+### Build
 To build the library and the provided samples run make in the build directory you executed CMake.
 
 `make`
 
+## Run
+After you complete the procedure above, you end up with the following sample applications in your build directory:
+
+* `kvsWebrtcClientMaster` - This application sends sample H264/Opus frames (path: `/samples/h264SampleFrames` and `/samples/opusSampleFrames`) via the signaling channel. It also accepts incoming audio, if enabled in the browser. When checked in the browser, it prints the metadata of the received audio packets in your terminal.
+* `kvsWebrtcClientViewer` - This application accepts sample H264/Opus frames and prints them out.
+* `kvsWebrtcClientMasterGstSample` - This application sends sample H264/Opus frames from a GStreamer pipeline.
+
+To run either of these samples, complete the following steps:
```

Comment:
It would be better to mention about the 4 samples execution after highlighting the set up steps below. Gives a more logical flow to the execution steps in my opinion.

Reply:
done!